aineko/core/node.py

```diff
@@ -120,35 +122,61 @@ def setup_datasets(
                 prefix
         """
         inputs = inputs or []
-        self.consumers.update(
+        # initialize the datasets:
+        self.inputs.update(
             {
-                dataset_name: DatasetConsumer(
-                    dataset_name=dataset_name,
-                    node_name=self.name,
-                    pipeline_name=self.pipeline_name,
-                    dataset_config=datasets.get(dataset_name, {}),
-                    prefix=prefix,
-                    has_pipeline_prefix=has_pipeline_prefix,
+                dataset_name: AbstractDataset.from_config(
+                    name=dataset_name, config=datasets.get(dataset_name, {})
                 )
                 for dataset_name in inputs
             }
         )
 
         outputs = outputs or []
-        self.producers.update(
+        self.outputs.update(
             {
-                dataset_name: DatasetProducer(
-                    dataset_name=dataset_name,
-                    node_name=self.name,
-                    pipeline_name=self.pipeline_name,
-                    dataset_config=datasets.get(dataset_name, {}),
-                    prefix=prefix,
-                    has_pipeline_prefix=has_pipeline_prefix,
+                dataset_name: AbstractDataset.from_config(
+                    name=dataset_name, config=datasets.get(dataset_name, {})
                 )
                 for dataset_name in outputs
             }
         )
 
+        for dataset_name in inputs:
+            if self.inputs[dataset_name].type == "kafka":
+                consumer_params = ConsumerParams(
+                    **{
+                        "dataset_name": dataset_name,
+                        "node_name": self.name,
+                        "pipeline_name": self.pipeline_name,
+                        "prefix": prefix,
+                        "has_pipeline_prefix": has_pipeline_prefix,
+                        "consumer_config": DEFAULT_KAFKA_CONFIG.get(
+                            "CONSUMER_CONFIG"
+                        ),
+                    }
+                )
+                self.inputs[dataset_name].initialize(
+                    create_consumer=True, connection_params=consumer_params
+                )
```

Comment:
This needs to be a separate function so we can initialize different dataset types here (and in the runner). This is fine if it will be refactored in a subsequent PR